packages/legend-application-studio/src/stores/EditorGraphState.ts

```diff
@@ -594,33 +662,45 @@ export class EditorGraphState {
       if (options?.openConsole) {
         this.editorStore.setActiveAuxPanelMode(AUX_PANEL_MODE.CONSOLE);
       }
-      const entities =
-        (yield this.editorStore.graphManagerState.graphManager.compileText(
+
+      const entitiesAndWarnings =
+        (yield this.editorStore.graphManagerState.graphManager.compileTextWithEntitiesAndWarnings(
           this.editorStore.grammarTextEditorState.graphGrammarText,
           this.editorStore.graphManagerState.graph,
-        )) as Entity[];
+        )) as EntitiesAndWarnings;
+      const entities = entitiesAndWarnings.entities;
 
-      if (!options?.disableNotificationOnSuccess) {
-        this.editorStore.applicationStore.notifySuccess(
-          'Compiled successfully',
+      const compilationWarnings = entitiesAndWarnings.engineWarnings;
+      this.editorStore.graphState.setIsStaleTextWarnings(false);
+
+      if (compilationWarnings && compilationWarnings.length > 0) {
+        this.editorStore.applicationStore.log.warn(
+          LogEvent.create(GRAPH_MANAGER_EVENT.COMPILATION_FAILURE),
+          'Compilation suceeded with warnings',
         );
+        this.editorStore.applicationStore.notifyWarning(
+          'Compilation suceeded with warnings',
+        );
+      } else {
+        if (!options?.disableNotificationOnSuccess) {
+          this.editorStore.applicationStore.notifySuccess(
+            'Compiled successfully',
+          );
+        }
       }
 
+      this.editorStore.graphState.setWarnings(
+        compilationWarnings as EngineWarning[],
+      );
+
       yield flowResult(this.updateGraphAndApplication(entities));
     } catch (error) {
       assertErrorThrown(error);
       if (error instanceof EngineError) {
-        this.editorStore.grammarTextEditorState.setError(error);
+        this.editorStore.grammarTextEditorState.setLocationForceCursor(error);
```

Comment:
let's revert the change you make to `error` in `GrammarTextEditorState` though, we still want to keep the error around. We can't coalesce it with warnings like what we're doing now.